pandas/tests/io/test_pytables.py

```diff
@@ -4552,6 +4552,20 @@ def test_legacy_table_fixed_format_read_py2(self, datapath):
                                                    name='INDEX_NAME'))
             assert_frame_equal(expected, result)
 
+    def test_legacy_table_read_py2(self, datapath):
+        # issue: 24925
+        # legacy table written in Python 2
+        with ensure_clean_store(
+                datapath('io', 'data', 'legacy_hdf',
+                         'legacy_table_py2.h5'),
+                mode='r') as store:
+            result = store.select('table')
+            expected = pd.DataFrame({
```

Comment:
Construct expected and do the assertion outside of the `ensure_clean_store` block.

Reply:
changed in https://github.com/pandas-dev/pandas/pull/25058/commits/0fa7102f96fde9b2f70a2cddb070ff973963d940